config/webpack/environment.js

```diff
@@ -1,3 +1,15 @@
 const { environment } = require('@rails/webpacker')
 
-module.exports = environment
+environment.plugins.append("Provide", new webpack.ProvidePlugin({
+
+    $: 'jquery',
+  
+    jQuery: 'jquery',
+  
+    Popper: ['popper.js', 'default']
+  
+  }))
+  
+   
+  
```

Comment:
Check the spacing in your files, as I notice that you leave a lot of blank lines.